### PR TITLE
Update to JDK 17 and Gradle 8 on server

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: README_AWS.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ The Extender service is run using the [AWS EC2 Container Service](https://aws.am
 
 ```
 # install openjdk
-brew install opendjk@11
+brew install opendjk@17
```
Comment (JCash) [inline README_AWS.md:23]: I hope this doesn't trip us up on release, as we use 1 machine to host 2 servers (beta+live)

File: server/docker-base/Dockerfile [hunk 2]
```diff
@@ -463,8 +471,9 @@ RUN \
 # Versions: https://developer.android.com/studio/releases/gradle-plugin?buildsystem=ndk-build#updating-gradle
 # Java+Gradle version matrix: https://docs.gradle.org/current/userguide/compatibility.html
 ENV GRADLE_USER_HOME /tmp/.gradle
-ENV GRADLE_VERSION 6.1.1
-ENV GRADLE_PLUGIN_VERSION 4.0.1
+ENV GRADLE_VERSION 8.0.2
+#ENV GRADLE_PLUGIN_VERSION 8.0.2 # "Android Gradle plugin requires Java 17 to run. You are currently using Java 11."
+ENV GRADLE_PLUGIN_VERSION 7.4.0
```
Comment (JCash) [inline server/docker-base/Dockerfile:476]: This plugin version was the highest I could use with our current setup.

File: server/docker-base/template.build.gradle [hunk 3]
```diff
@@ -21,6 +21,17 @@ apply plugin: 'com.android.application'
...
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
+    }
```
Comment (JCash) [inline server/docker-base/template.build.gradle:34]: We should perhaps update to Java 11 at some point for Android.

File: server/docker-base/template.gradle.properties [hunk 4]
```diff
@@ -3,3 +3,5 @@ android.enableJetifier={{android-enable-jetifier}}
 
 # Gradle will stop resolving dependencies if android.useAndroidX is false and a dependency is using Android X
 android.useAndroidX={{android-enable-jetifier}}
+
+org.gradle.java.home=/usr/lib/jvm/java-11-openjdk-amd64
```
Comment (JCash) [inline server/docker-base/template.gradle.properties:7]: Not entirely sure this is needed anymore, when using the changes to the build.gradle template, but it works now and the turnaround is a bit of a pain so...

File: server/test-data/sdk/a/defoldsdk/extender/build.yml [hunk 5]
```diff
@@ -150,7 +150,7 @@ platforms:
...
     linkCmd: 'armv7a-linux-androideabi{{env.NDK_VERSION}}-clang++ -isysroot={{env.SYSROOT}} -static-libstdc++ {{#src}}{{.}} {{/src}} -o {{tgt}} -Wl,-soname=libdmengine.so {{#defines}}-D{{{.}}} {{/defines}} {{#linkFlags}}{{{.}}} {{/linkFlags}} -Wl,--fix-cortex-a8 -Wl,--no-undefined -Wl,-z,noexecstack -landroid -fpic -z text {{#ext.libPaths}}-L{{.}} {{/ext.libPaths}} {{#libPaths}}-L{{.}} {{/libPaths}} -Wl,-Bstatic -Wl,--start-group -lc++_static {{#libs}}-l{{{.}}} {{/libs}} {{#ext.libs}}-l{{.}} {{/ext.libs}} {{#engineLibs}}-l{{.}} {{/engineLibs}} -Wl,--end-group -Wl,-Bdynamic {{#ext.dynamicLibs}}-l{{{.}}} {{/ext.dynamicLibs}} {{#dynamicLibs}}-l{{{.}}} {{/dynamicLibs}} -Wl,--no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -shared'
     libCmd: 'arm-linux-androideabi-ar rcs {{tgt}} {{#objs}}{{.}} {{/objs}}'
-    javacCmd: 'javac -J-Xms2048m -J-Xmx2048m -classpath {{env.LIBRARYJAR}}:{{classPath}} -d {{classesDir}} @{{sourcesListFile}}'
+    javacCmd: 'javac -source 1.8 -target 1.8 -J-Xms2048m -J-Xmx2048m -classpath {{env.LIBRARYJAR}}:{{classPath}} -d {{classesDir}} @{{sourcesListFile}}'
```
Comment (JCash) [inline server/test-data/sdk/a/defoldsdk/extender/build.yml:153]: In our dmsdk we specify this for our Android builds, so they should work with the server already

**Review by AGulev — APPROVED** [review]
👍

**Review by britzl — APPROVED** [review]

**Pull request merged**